### ISSUE TITLE
Slang interpreter: interp_2 crashes when printing the heap

### ISSUE THREAD

**Issue by Camak2441** [yallop/cc_cl_cam_ac_uk#21]
The bounds check when printing the heap is incorrect. It is currently
```OCaml
if i < k 
    then "" 
    else (string_of_int k) ^ " -> " ^ (string_of_value (IntMap.find k heap)) ^ "\n" ^ (aux (k+1))
```
which causes it to attempt to find the kth element of the heap, throwing a `Not_found` exception. 

It should be

```OCaml
if i <= k 
    then "" 
    else (string_of_int k) ^ " -> " ^ (string_of_value (IntMap.find k heap)) ^ "\n" ^ (aux (k+1))
```

**Comment by yallop** [comment]
Thank you for the report.  Here's a reproduction:

Before:

```
$ ./_build/default/slang/slang.exe -v  -i2 /dev/stdin <<< 'ref 1'
Compile code =
[PUSH 1;
 MK_REF]

State 1 : 
Code Stack = 
[PUSH 1;
 MK_REF]
Env/Value Stack = 
[]

State 2 : 
Code Stack = 
[MK_REF]
Env/Value Stack = 
[V 1]

ERROR in /dev/stdin with Interpreter 2 : Exception: Not_found
```


After:
```
$ ./_build/default/slang/slang.exe -v  -i2 /dev/stdin <<< 'ref 1'
Compile code =
[PUSH 1;
 MK_REF]

State 1 : 
Code Stack = 
[PUSH 1;
 MK_REF]
Env/Value Stack = 
[]

State 2 : 
Code Stack = 
[MK_REF]
Env/Value Stack = 
[V 1]

State 3 : 
Code Stack = 
[]
Env/Value Stack = 
[V REF(0)]
Heap = 
0 -> 1


Interpreter 2 
output> REF(0)
```

**Comment by yallop** [comment]
Fixed in 634d699eef95ce874dcdedf3b91d233827c3790c.
